CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs

```diff
@@ -0,0 +1,293 @@
+        {
+            //Arrange
+            HomeworkRequestDto homeworkRequestDto = new HomeworkRequestDto()
+            {
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z")
+            };
+
+            _lessonRepositoryMock.Setup(l => l.IsEntityExistAsync(1)).ReturnsAsync(true);
+            _homeworkRepositoryMock.Setup(x => x.GetByIdAsync(It.IsAny<long>())).ReturnsAsync(default(Homework));
+
+            _unitOfWorkMock.Setup(x => x.HomeworkRepository).Returns(_homeworkRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+            //Act
+
+            var result = await _homeworkService.UpdateHomeworkAsync(1, homeworkRequestDto);
+
+            //Assert
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.NotFound);
+
+        }
+
+        [Fact]
+        public async Task UpdateHomeworkAsync_ValidData_ShouldReturnValidationError()
+        {
+            //Arrange
+            HomeworkRequestDto homeworkRequestDto = new HomeworkRequestDto()
+            {
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z")
+            };
+
+            Homework homework = new Homework
+            {
+                Id = 1,
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z"),
+
+            };
+
+            _lessonRepositoryMock.Setup(l => l.IsEntityExistAsync(1)).ReturnsAsync(true);
+            _homeworkRepositoryMock.Setup(x => x.GetByIdAsync(It.IsAny<long>())).ReturnsAsync(homework);
+
+            _unitOfWorkMock.Setup(x => x.HomeworkRepository).Returns(_homeworkRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+            //Act
+
+            var result = await _homeworkService.UpdateHomeworkAsync(1, homeworkRequestDto);
+
+            //Assert
+            result.Data.Should().BeEquivalentTo(new HomeworkDto
+            {
+                Id = 1,
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z"),
+                AttachmentIds = new List<long>()
+            });
+
+        }
+
+        #endregion
+
+        #region GetHomework
+
+        [Fact]
+        public async Task GetHomeworkByLessonIdAsync_DefaultLessonId_ShouldReturnValidationError()
+        {
+            //Arrange
+            long lessonId = default;
+
+            //Act
+            var result = await _homeworkService.GetHomeworksByLessonId(lessonId);
+
+            //Assert
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        [Fact]
+        public async Task GetHomeworkByLessonIdAsync_NotExistingLessonId_ShouldReturnEmptyListOfHomework()
+        {
+            //Arrange
+            long lessonId = 1;
```

Comment:
Perhaps this variable should be named `nonExistentLessonId`
Please update everywhere else too.

Reply:
Done